content/api/vision/coordinates.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Coordinates
+---
+
+[Polar coordinates](https://en.wikipedia.org/wiki/Polar_coordinate_system) represent a distance from a set point and an angle from a set direction. [Cartesian coordinates](https://en.wikipedia.org/wiki/Cartesian_coordinate_system) use three perpendicular axes (typically _x_, _y_ and _z_) to specify a point in space.
+
+{{% notice note %}}
+All distances are measured in metres.
+{{% /notice %}}
+
+## Polar Coordinates
+The polar coordinates of a marker are relative to the view of the camera.
+
+ - `distance_metres` - the distance from the camera to the marker.
+ - `rot_x_rad` / `rot_x_deg` - the angle of rotation in radians/degrees about the x axis. The x axis runs horizontally in front of the camera. A positive angle would be one where the marker is in front of the camera.
+ - `rot_y_rad` / `rot_y_deg` - the angle of rotation in radians/degrees about the y axis. The y axis runs vertically in front of the camera. A positive angle would be one where the marker is rotated clockwise about the y axis looking from the top of the marker.
+ 
+{{% notice info %}}
+For example, if `distance_metres` was `1`, `rot_x_deg` was `0`, and `rot_y_deg` was `45`, the marker would be 45 degrees to the right of the robot, exactly 1 metre away.
```

Comment:
This example matches my expectation of the polar coordinates (and what I've implemented in https://github.com/sourcebots/sb-vision/pull/23), however is inconsistent with what's stated above.

In the above system, the `rot_x` value here makes the `rot_y` value meaningless as 0° is described as being above the robot.

Alternatively in the above system, the description here would instead be `Polar(rot_x_deg=90, rot_y_deg=45, distance_meters=1)`.

Reply:
I think more testing (or code reading) is required to work out exactly what value of `rot_y` is directly infront of the robot at webcam level

Reply:
I don't think that testing is the right approach -- this is something we can define; we just need to work out what we want it to be. I'll have another read of the current polar implementation, though my current understanding is that it neither matches any of these nor even makes sense.

Having read https://en.wikipedia.org/wiki/Spherical_coordinate_system some more, It looks like we're unfortunately not using a standard anyway; we should have `rot_x` (_approximately_ θ in the ISO/physics model) and `rot_z` (φ) if we were. However even that is inaccurate since `rot_x` diverges from `θ` for large values of `rrot_z`.

If we change this at all, we're technically making a breaking change. I'm therefore tempted to suggest that we should break as _loudly_ as possible if we do need to break things. I see the following options:
 - drop polar coordinates entirely for now. They're not hard to work out, but they are hard to describe/document.
 - adopt the physics/ISO convention and change the property names (also rename to `spherical` rather than `polar`)
 - make no changes to the functionality, just document what's there
 - make no changes to the functionality, but remove the polar docs

For this PR I'm most tempted to just remove the polar docs, so that we get the other tidyups sooner, and then come back to this later.

Reply:
Ok, so following a little investigation, I've created some diagrams which capture my understanding of the current state of things. (I've got them as SVGs so we could even adapt them to be useful in the docs if we wanted too).

### Current implementation
Here the `polar_x` and `polar_y` values to match the output from the current [`cartesian_to_polar`](https://github.com/sourcebots/sb-vision/blob/57675c144b89eedb7ef451483c5a69412bc44d4a/sb_vision/tokens.py#L161); they translate to `rot_x...` and `rot_y...` respectively on the `robot-api` side.

These angles are not really in the right pace to be used (they're the wrong side of the triangle for one), though they also express a "rotation along the {x,y} axis" rather than "rotation about the {x,y} axis". Essentially I think that `polar_x` was intending to be "What's the angle of turn equivalent to the Cartesian 'x' value?"

![vision-reference-current-polar-implementation](https://user-images.githubusercontent.com/336212/35685904-e46f3fe6-0762-11e8-92e2-47efc14f844e.png)

### Docs inline description
The angles here are angles _about_ an axis, which makes more sense than the current implementation.

This measures `rot_x` down from vertical, rather than up from horizontal.

![vision-reference-docs-inline](https://user-images.githubusercontent.com/336212/35685919-eef5b486-0762-11e8-9565-0ad4687f2315.png)

### Docs example
The angles here are (again) angles _about_ an axis, which makes more sense than the current implementation.

This measures `rot_x` up from horizontal, rather than down from vertical, and feels more natural to me.

![vision-reference-docs-example](https://user-images.githubusercontent.com/336212/35685925-f2bffffe-0762-11e8-9392-00dcdd314e59.png)

### Others
I've not done a diagram for the ISO spherical system as it doesn't decompose into two flat images well; the diagram on wikipedia is also pretty good.

Reply:
To me, the _Docs example_ setup makes the most sense. I think there's much room for improvement here, especially in making sure that what's implemented actually makes sense, and that it matches what's in the docs. 

These images are really nice, I think it might be better to use those over made-up examples to better explain what each measurement is of!

Reply:
> These images are really nice, I think it might be better to use those over made-up examples to better explain what each measurement is of!

Thanks! I think it's worth keeping the written example as well -- this is a very complicated topic, so having lots of ways it's explained/demonstrated is a good thing.

------

The more I think about this I feel we shouldn't change the existing behaviour for those who are using it. While it's clearly not the best, I think it would be usable. Hence if people have already started using it they'll likely have figured out how it behaves by testing.

I suggest that we:
1. drop the docs for the `polar` behaviour entirely (in this PR); maybe mention that it exists but is broken
2. (separately) introduce a `spherical` attribute and document that, with documentation largely lifted from this PR

This avoids breaking anyone already using the `polar` support, while allowing us to provide a better solution.

Thoughts?

Reply:
Due to the fact vision is almost certainly broken (https://github.com/sourcebots/sb-vision/pull/22) I personally think it's perfectly ok to drop `polar` entirely. I'd rather fully deprecate an api (perhaps with a print and a docs link), than silently replace it and remove its docs. We just need to get the `spherical` system in.

Reply:
(context: since my above comment I've since done most of the work for my proposed approach; I'm open to abandoning it if we feel that if another approach is better though)

@RealOrangeOne note that at the moment the vision system is using the `'c270'` calibration data, so it should be wrong but somewhat consistently so -- we never actually shipped the tecknet support.